app/src/main/java/org/schabi/newpipe/CheckForNewAppVersionTask.java

```diff
@@ -0,0 +1,230 @@
+package org.schabi.newpipe;
+
+import android.app.Application;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.Signature;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.preference.PreferenceManager;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.NotificationManagerCompat;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.schabi.newpipe.report.ErrorActivity;
+import org.schabi.newpipe.report.UserAction;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * AsyncTask to check if there is a newer version of the NewPipe github apk available or not.
+ * If there is a newer version we show a notification, informing the user. On tapping
+ * the notification, the user will be directed to the download link.
+ */
+public class CheckForNewAppVersionTask extends AsyncTask<Void, Void, String> {
+
+    private static final Application app = App.getApp();
+    private static final String GITHUB_APK_SHA1 = "B0:2E:90:7C:1C:D6:FC:57:C3:35:F0:88:D0:8F:50:5F:94:E4:D2:15";
+    private static final String newPipeApiUrl = "https://newpipe.schabi.org/api/data.json";
+    private static final int timeoutPeriod = 30;
+
+    private SharedPreferences mPrefs;
+    private OkHttpClient client;
+
+    @Override
+    protected void onPreExecute() {
+
+        mPrefs = PreferenceManager.getDefaultSharedPreferences(app);
+
+        // Check if user has enabled/ disabled update checking
+        // and if the current apk is a github one or not.
+        if (!mPrefs.getBoolean(app.getString(R.string.update_app_key), true)
+                || !isGithubApk()) {
+            this.cancel(true);
+        }
```

Comment:
Checking this before starting the task would be more efficient, at App#L108

Reply:
This is 4 years old code, is this still the case?

Reply:
Hi @Stypox,

I apologize for my previous comment. I didn't realize that the code was 4 years old, and I see that it has now been deleted. I was going through the code to explore and understand the codebase, but I should have been more careful about commenting on old code.

I'll be more mindful of the age of code in the future and avoid commenting on old code unless I'm sure it's still relevant.

Thanks for your patience and understanding.